swift-sdk/Internal/IterableHtmlMessageViewController.swift

```diff
@@ -17,18 +17,18 @@ class IterableHtmlMessageViewController: UIViewController {
     struct Parameters {
         let html: String
         let padding: UIEdgeInsets
-        let callback: ITBURLCallback?
+        let result: Promise<URL, IterableError>
```

Comment:
Invert the control flow here: instead of taking `result` as a parameter, return it as a result of `show`, etc.